gensim/test/test_word2vec2tensor.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (C) 2018 Aakaash Rao   <aakaash@uchicago.edu>
+# Copyright (C) 2018 Radim Rehurek <radimrehurek@seznam.cz>
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+"""Test for gensim.scripts.word2vec2tensor.py."""
+
+import logging
+import unittest
+import sys
+
+import numpy
+import gensim
+
+from gensim.utils import check_output, smart_open
+from gensim.test.utils import get_tmpfile
+
+
+class TestWord2Vec2Tensor(unittest.TestCase):
+    def setUp(self):
+        # self.datapath = datapath('test_word2vec.txt')
+        self.datapath = 'test/test_data/test_word2vec.txt'  # temporary
+        self.output_folder = get_tmpfile('')
+        self.metadata_file = self.output_folder + '_metadata.tsv'
+        self.tensor_file = self.output_folder + '_tensor.tsv'
+        self.vector_file = self.output_folder + '_vector.tsv'
+
+    def testConversion(self):
+        check_output(args=[
+            sys.executable, '-m', 'scripts.word2vec2tensor', '--input',
+            self.datapath, '--output', self.output_folder
+        ])
+        try:
+            with smart_open(self.metadata_file, 'rb') as f:
+                self.metadata = f.readlines()
+        except FileNotFoundError:
+            self.fail(
+                'Metadata file %s creation failed. \
+                Check the parameters and input file format.' % self.metadata_file
+                )
+        try:
+            with smart_open(self.tensor_file, 'rb') as f:
+                self.vectors = f.readlines()
```

Comment:
I don't catch, why you need `self` here (same question for `number_words`, `metadata`, etc)?